### DISCUSSION TITLE
Is there a way to preserve the order of elements in an object?

### DISCUSSION THREAD

**Discussion by al-martyn1** [nlohmann/json#3438] [Q&A] [answered]
I want that after parsing the input JSON, the fields in the object would be traversed not in lexicographic order, but in the order in which they occurred in the source file?

This is very convenient when the config JSON is written manually, but can also be updated programmatically. In this case, the order of the elements is lost, and it is difficult for a person to navigate in it.

Also, when adding elements programmatically, I would like them to be serialized in the order of addition/insertion, and so that it would be possible to insert the 'tags' field after 'info' between the 'openapi', 'info', 'paths', 'components' fields and get the order: 'openapi', 'info', 'tags', 'paths', 'components'. Now, after serialization, we get 'components', 'info', 'openapi', 'paths', 'tags', which is very inconvenient - the fields 'openapi', 'info', 'tags' are information headers, and should be at beginning.

Using SAX is not very suitable, because there is already a lot of code written for JSON DOM, and writting my own version of DOM is not possible.

I believe that this can slow down the processing of JSON files, and maybe such behavior could be turned on and off with a template  parameter or some other option?

**Comment by falbrechtskirchinger** [comment, marked as answer]
This should be what you're asking for: https://json.nlohmann.me/features/object_order/#alternative-behavior-preserve-insertion-order

Example: https://godbolt.org/z/nMG7so8c7
